src/image/p5.Image.js

```diff
@@ -220,6 +220,40 @@ p5.Image = class {
     this.pixels = [];
   }
 
+  /**
+   * Set the pixel density of the image.
+   *
+   * @method setPixelDensity
+   * @param {Number} density - The pixel density to set.
+   * @example
+   * <div><code>
+   * let img = new p5.Image(100, 100);
+   * img.setPixelDensity(2);
+   * </code></div>
+   */
+  setPixelDensity(density) {
+    if (density <= 0) {
+      const errorObj = {
+        type: 'INVALID_VALUE',
+        format: { types: ['Number'] },
+        position: 1,
+      };
+
+      p5._friendlyParamError(errorObj, 'setPixelDensity');
+
+      // Default to 1 in case of an invalid value
+      density = 1;
+    }
+
+    this._pixelDensity = density;
+
+    // Adjust canvas dimensions based on pixel density
+    this.canvas.width = this.width * density;
+    this.canvas.height = this.height * density;
+
+    // Update the drawing context
+    this.drawingContext = this.canvas.getContext('2d');
```

Comment:
I think it might not be necessary to get the context again after changing the canvas size